awpy/types.py

```diff
@@ -665,14 +665,81 @@ class RoundStatistics(TypedDict):
     players_killed: dict[Literal["CT", "T"], set[str]]
 
 
-class FrameMapControl(TypedDict):
-    """Type for the result of awpy.analytics.map_control.calc_map_control.
+# Type to represent tile id for navigation tiles.
+# TODO: Incorporate TileIdObject into rest of repo.
+TileIdObject = NewType("TileIdObject", int)
 
-    Holds map control values for each time for a certain frame
+# eturn type for awpy.analytics.map_control._bfs_helper.
+# Contains map control values for one team.
+# Maps TileIdObject to list of tile map control values.
+TeamMapControlValuesDict = NewType(
+    "TeamMapControlValuesDict", dict[TileIdObject, list[float]]
+)
+
+# Return type for awpy.analytics.map_control.graph_to_tile_neighbors
+# Maps TileIdObject to set of neighboring tiles.
+TileNeighborsDict = NewType("TileNeighborsDict", dict[TileIdObject, set[int]])
+
+
+class TileDistanceObject(TypedDict):
+    """Dataclass with data for map control tile distance calculations.
+
+    Holds information for distance to source tile and tile_id
+    distance is associated with.
+
+    TODO: Merge TileDistanceObject with TileDistance object
+          as their functionality seems very similar
+    """
+
+    distance: float
+    tile_id: TileIdObject
+
+
+# @dataclass
+class BFSTileData(TypedDict):
+    """Dataclass containing data for tiles during bfs algorithm.
+
+    Holds information for tile_id for tile, current map control
+    value, and steps remaining for bfs algorithm
+    """
+
+    tile_id: TileIdObject
+    map_control_value: float
+    steps_left: int
+
+
+# @dataclass
+class FrameTeamMetadataDict(TypedDict):
+    """Dataclass containing metadata for one team.
+
+    Holds information for aliver player locations. Can include
+    more metadata (utility, bomb location, etc.) in the future
+    """
+
+    alive_player_locations: list[list[float]] | list
```

Comment:
This thread is also still open.

Reply:
Oop my bad. Addressed in next commit to branch (introduced `PlayerPosition` TypeAlias for `list[float]`. Thanks!